src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/LanguageServiceHost.cs

```diff
@@ -94,6 +99,45 @@ private Task OnProjectFactoryCompletedAsync()
 
             // Update project context and subscriptions.
             await UpdateProjectContextAndSubscriptionsAsync().ConfigureAwait(false);
+            LogLanguageServiceHostInitializeStop();
+        }
+
+        private static volatile bool s_isFirstStartEvent = true;
+        private static volatile int s_numberOfTimesInitialized = 0;
+        private static object s_telemetryLock = new object();
+
+        /// <summary>
+        /// Only posts an event the first time this method is called in the VS process
+        /// Also updates a global count so we know how many project hosts have been initialized
+        /// </summary>
+        private void LogLanguageServiceHostInitializeStart()
+        {
+            if (s_isFirstStartEvent)
```

Comment:
Couldn't you avoid the use of the lock and bool by using the return of the Interlocked.Increment being 1?